packit/config/package_config.py

```diff
@@ -158,6 +159,26 @@ def get_from_dict(
 
         return PackageConfigSchema().load(raw_dict)
 
+    @staticmethod
+    def validate_file_set_and_present(
+        raw_dict: dict, item_key: str, file_path: Union[str, None]
+    ) -> None:
+        """
+        Checks if a file path is set(not none)
+        and if the file itself is present under
+        the specified path. Throws a
+        PackitConfigException if the validation
+        fails.
+        """
+        if file_path:
+            if os.path.isfile(file_path):
+                raw_dict[item_key] = file_path
+            else:
+                raise PackitConfigException("Spec file not found under " + file_path)
```

Comment:
Ok Im probably just being dumb, could you please explain why is this statement unreachable? :D

Reply:
No problem, I had to look it up too through the references :) The `file_path` you pass in to the function is the `spec_file_path` variable and that one gets initialized:

1. by globbing through the repo
2. querying git-forge (GitHub, GitLab or Pagure) API

^ That means you either get a valid path or `None`.
If you get `None` you fall into the outer `else` block and if you get path, it is always valid, otherwise you wouldn't be able to find it.

Reply:
Ah ok cool, that explains it, thanks :)